### PR TITLE
Merge 2017-01-17

### PR DESCRIPTION
**Pull request by maxim-belkin**

### CODE REVIEW

File: Formula/ghc.rb [hunk 1]
```diff
@@ -75,27 +62,17 @@ class Ghc < Formula
...
   resource "binary" do
     if OS.linux?
-      # Using 8.0.1 gives the error message:
-      # strip: Not enough room for program headers, try linking with -N
-      url "http://downloads.haskell.org/~ghc/7.8.4/ghc-7.8.4-x86_64-unknown-linux-deb7.tar.xz"
-      sha256 "f62e00e93a5ac16ebfe97cd7cb8cde6c6f3156073d4918620542be3e0ad55f8d"
-    elsif MacOS.version >= :sierra
-      url "https://downloads.haskell.org/~ghc/8.0.2-rc2/ghc-8.0.1.20161213-x86_64-apple-darwin.tar.xz"
-      sha256 "f5cacbf6ae9394cfd191244e6a7ab385184c6dc5168cb2c576ab2d74543c1391"
+      url "https://downloads.haskell.org/~ghc/8.0.2/ghc-8.0.2-x86_64-deb8-linux.tar.xz"
```
Comment (rwhogg) [inline Formula/ghc.rb]: I'm not familiar with GHC; does 8.0.2 resolve the previous problems we were having with 8.0.1?

File: Formula/tty-clock.rb [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+    sha256 "9341fb07070b665dc5f9593c1b4811ec734f7221afbde2547cee55fc9102aa1e" => :yosemite
   end
 
+  depends_on "pkg-config" => :build
+
   def install
-    inreplace "Makefile", "/usr/local/bin/", "#{bin}/"
-    system "make"
-    system "make", "install"
+    ENV.append "LDFLAGS", "-lncurses"
```
Comment (rwhogg) [inline Formula/tty-clock.rb]: This formula looks like it needs to depend on `homebrew/dupes/ncurses` on Linux.

Comment (rwhogg) [inline Formula/tty-clock.rb]: Hmm, after upgrading I got some strange errors on this formula:

```
In file included from ttyclock.h:44:0,
                 from ttyclock.c:33:
/home/linuxbrew/.linuxbrew/include/ncursesw/unctrl.h:60:38: error: conflicting types for 'unctrl'
 NCURSES_EXPORT(NCURSES_CONST char *) NCURSES_SP_NAME(unctrl) (SCREEN*, chtype);
                                      ^
In file included from /usr/include/ncursesw/ncurses.h:2012:0,
                 from ttyclock.h:44,
                 from ttyclock.c:33:
/home/linuxbrew/.linuxbrew/include/ncursesw/unctrl.h:57:38: note: previous declaration of 'unctrl' was here
 NCURSES_EXPORT(NCURSES_CONST char *) unctrl (chtype);
                                      ^
make: *** [Makefile:26: tty-clock] Error 1
```

Same thing seems to happen on a fresh install.

Comment (maxim-belkin) [inline Formula/tty-clock.rb]: I was able to build tty-clock just fine in just 4 seconds ([here](https://gist.github.com/753e72bd3044e7fe1602157a7b1942dd))
```bash
$ patchelf --print-needed /path/to/tty-clock
libncursesw.so.6
libc.so.6
```

Comment (maxim-belkin) [inline Formula/tty-clock.rb]: this is with your fix for the extra dependency on `ncurses`

File: Formula/ghc.rb (hunk 1 above)
Comment (maxim-belkin) [inline Formula/ghc.rb]: Don't know decided to give it a try but it failes for me locally so we might be better off leaving this change out.

**Comment by maxim-belkin** [comment]
Closing this one: will do another merge.

**Pull request closed without merging**